File: src/main/java/com/example/examplemod/ExampleModJaJpLanguageProvider.java
package com.example.examplemod;

import net.minecraft.data.DataGenerator;
import net.minecraft.data.IDataProvider;
import net.minecraftforge.common.data.LanguageProvider;

public class ExampleModJaJpLanguageProvider  extends LanguageProvider {
    public ExampleModJaJpLanguageProvider(DataGenerator gen, String modId) {
        super(gen, modId, "ja_jp");
    }

    @Override
    protected void addTranslations() {
        add(Blocks.TITANIUM_BLOCK.get(), "チタンブロック");
        add(Items.TITANIUM_INGOT.get(), "チタンインゴット");
    }
}

File: src/main/java/com/example/examplemod/ExampleModJaJpLanguageProvider.kt
package com.example.examplemod

import net.minecraft.data.DataGenerator
import net.minecraftforge.common.data.LanguageProvider

class ExampleModJaJpLanguageProvider(gen: DataGenerator, modId: String) : LanguageProvider(gen, modId, "ja_jp") {
    override fun addTranslations() {
        add(ExampleBlocks.TITANIUM_BLOCK.get(), "チタンブロック")
        add(ExampleItems.TITANIUM_INGOT.get(), "チタンインゴット")
    }
}
